library/fluid/fluid_extensions/src/main/java/org/quiltmc/qsl/fluid/api/QuiltDirectionalFluid.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.fluid.api;
+
+
+import com.google.common.collect.Maps;
+import com.mojang.datafixers.util.Pair;
+import it.unimi.dsi.fastutil.shorts.Short2BooleanMap;
+import it.unimi.dsi.fastutil.shorts.Short2BooleanOpenHashMap;
+import it.unimi.dsi.fastutil.shorts.Short2ObjectMap;
+import it.unimi.dsi.fastutil.shorts.Short2ObjectOpenHashMap;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Material;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.shape.VoxelShape;
+import net.minecraft.util.shape.VoxelShapes;
+import net.minecraft.world.BlockView;
+import net.minecraft.world.WorldAccess;
+import net.minecraft.world.WorldView;
+
+import java.util.EnumMap;
+import java.util.Map;
+
+/**
+ * @apiNote Used when u want to specify custom fluid flow direction, otherwise use QuiltFluid
+ */
+public abstract class QuiltDirectionalFluid extends QuiltFluid implements QuiltFlowableFluidExtensions{
+
+	public Direction getFlowingDirection() {return Direction.DOWN;}
```

Comment:
```suggestion
	public Direction getFlowingDirection() {
		return Direction.DOWN;
	}
```
two comments for this one:
I think `getFlowDirection` would be a nicer name  
this method also needs some documentation, even though it's always getting overridden